### PR TITLE
[BUGFIX] handle multibyte characters in iCal description

### PR DESCRIPTION
**Pull request by rr-it**
Without this fix the iCal description is not multibyte character aware.

This results in multibyte character-splitting at line ends:

```text
  bei schwerer betroffenen Erkrankten viel Positives bewirken und ein gröÍ
 tmögliches Maß an Beweglichkeit erhalten. Viele Beispiele und wissensch
```
By wrongly splitting multibyte chars, the Gnome Linux Desktop does not recognize the iCal entry and fails. On other systems e.g. Android with Google calendar the text is displayed broken.

---

As of [RFC5545 Section 3.1](https://www.rfc-editor.org/rfc/rfc5545#section-3.1):
"Lines of text SHOULD NOT be longer than 75 octets, excluding the line break."
– So having less octets/bytes is fine as well.

With multibyte characters I made this assumption for description text:
`In worst case there are a max of 2 bytes per UTF8-character - averaged per line.`
Therefor split to max 37 chars: roughly equals max 74 bytes (plus 1 byte for white-space).

For a more fine-grained solution see:
https://github.com/sabre-io/vobject/blob/06feff370141fd3118609f808e86d9315864bf14/lib/Property.php#L233-L241

Are we compatible to `http://sabre.io/license/ Modified BSD License` – and thereby may use the mentioned code?

---

For unit tests: Waiting for this solution to be selected favorable.

- [ ] Fix existing tests
- [ ] Add multibyte test-cases

### CODE REVIEW

**Comment by derhansen** [comment]
Thanks for your finding. I would however prefer to fix this as following:

```
Index: Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php b/Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php
--- a/Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php	(revision d44c5bae77b9a5b10a5a3f2b3de02fe0b4470a7a)
+++ b/Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php	(date 1646159506344)
@@ -53,9 +53,9 @@
         $tmpDescription = str_replace(chr(10), '', $tmpDescription);
         // Glue everything together, so every line is max 75 chars
         if (strlen($tmpDescription) > 75) {
-            $newDescription = substr($tmpDescription, 0, 63) . chr(10);
-            $tmpDescription = substr($tmpDescription, 63);
-            $arrPieces = str_split($tmpDescription, 74);
+            $newDescription = mb_substr($tmpDescription, 0, 63) . chr(10);
+            $tmpDescription = mb_substr($tmpDescription, 63);
+            $arrPieces = mb_str_split($tmpDescription, 74);
             foreach ($arrPieces as &$value) {
                 $value = ' ' . $value;
             }

```

I know this may exceed the 75 octets limit for UTF-8 chars, but the RFC states, that lines `SHOULD NOT` be actually they can be longer. I also checked several iCal exports (e.g. Google Calendar) and they also keep a 75 char limit and ignore the 75 octet limit.

So lets keep this change simple and fix it as I suggested. Will you change your PR or should I fix the bug?

File: Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php [hunk 1]
```diff
@@ -52,10 +52,10 @@ public function render(): string
         // Strip new lines
         $tmpDescription = str_replace(chr(10), '', $tmpDescription);
         // Glue everything together, so every line is max 75 chars
-        if (strlen($tmpDescription) > 75) {
-            $newDescription = substr($tmpDescription, 0, 63) . chr(10);
-            $tmpDescription = substr($tmpDescription, 63);
-            $arrPieces = str_split($tmpDescription, 74);
+        if (mb_strlen($tmpDescription) > 63) {
```
Comment (rr-it) [inline Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php:55]: Switching to `mb_strlen` for correct counting and max chars 63 for first line - like Google Calendar.

File: Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php [hunk 2]
```diff
@@ -52,10 +52,10 @@ public function render(): string
...
+            $newDescription = mb_substr($tmpDescription, 0, 63) . chr(10);
+            $tmpDescription = mb_substr($tmpDescription, 63);
+            $arrPieces = mb_str_split($tmpDescription, 75);
```
Comment (rr-it) [inline Classes/ViewHelpers/Format/ICalendarDescriptionViewHelper.php:58]: Switching to max chars 75 and afterwards an additional whitspace is added resulting in 76 chars -  like Google Calendar.

**Comment by derhansen** [comment]
Thanks. I'll merge this and push a follow up for the tests and make some other improvements

**Pull request merged**